src/app/components/ContractVerificationIcon/index.tsx

```diff
@@ -49,13 +55,13 @@ export const VerificationIcon: FC<{
   }
   const Component = noLink ? Box : (Link as React.ElementType)
   const componentProps = noLink ? {} : sourcifyLinkProps
-
+  const hideLabelMobile = hideLabel && isMobile
   return (
     <>
       <Component {...componentProps}>
-        <ContractStatus verificationLevel={verificationLevel} />
+        <ContractStatus verificationLevel={verificationLevel} hideLabel={hideLabelMobile} />
       </Component>
-      &nbsp; &nbsp;
+      {hideLabel && isMobile ? null : '\u00A0\u00A0'}
```

Comment:
I'll fix: spacing here is not related to label nor isMobile. It should be displayed depending on noLink